framework/src/postprocessors/ElementAverageMaterialProperty.C

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElementAverageMaterialProperty.h"
```

Comment:
Looks like we have a conflict in Rattlesnake! Do you have access? We should make sure that this class (and the other similar ones) have the same capabilities as the ones in Rattlesnake. Moving these up will be a bit of work, since it'll require patches to 4 downstream applications. Are you prepared for the pain and suffering?

Reply:
I noticed that... Their class has exactly the same interface but just is implemented differently. I wish they would have put it in MOOSE from the start! I'm ready for the suffering.

Reply:
(This is why I went on a shooting rampage in our pod, by the way)

Reply:
Yeah, we had an active shooter situation - kinda sorta ;-)

Reply:
![image](https://user-images.githubusercontent.com/3036960/73381079-653e8800-4282-11ea-9532-2b86304d710f.png)

Reply:
Developers tend to take an easier route. I think back to the day (I think before MOOSE becomes open source), adding that in Rattlesnake is way simpler than pushing it into MOOSE framework. I feel sorry to this ;-)

Reply:
@YaqiWang It happens sometimes. I already git blamed to find that it was Avery Bingham's fault (I wanted to see who to shoot with NERF gun), so I don't necessarily blame you ;)